worlds/ufo50/__init__.py

```diff
@@ -64,30 +79,72 @@ class UFO50World(World):
     settings_key = "ufo_50_settings"
     settings: ClassVar[UFO50Settings]
 
-    manager: GameManager
+    using_ut: bool
+    ut_passthrough: Dict[str, Any]
+
+    included_games: List  # list of modules for the games that are going to be played by this player
 
     def generate_early(self) -> None:
-        self.manager = GameManager(self)
         if not self.player_name.isascii():
             raise OptionError(f"{self.player_name}'s name must be only ASCII.")
 
+        if hasattr(self.multiworld, "re_gen_passthrough"):
+            if "UFO 50" in self.multiworld.re_gen_passthrough:
+                self.ut_passthrough = self.multiworld.re_gen_passthrough["UFO 50"]
+                self.options.always_on_games.value = self.ut_passthrough["included_games"]
+                self.options.random_choice_games.value.clear()
+                self.options.random_choice_game_count.value = 0
+
+        included_game_names = sorted(self.options.always_on_games.value)
+        # exclude always on games from random choice games
+        maybe_games = sorted(self.options.random_choice_games.value - self.options.always_on_games.value)
+        # if the number of games you want is higher than the number of games you chose, enable all chosen
+        if self.options.random_choice_game_count >= len(maybe_games):
+            included_game_names += maybe_games
+        elif self.options.random_choice_game_count and maybe_games:
+            included_game_names += self.random.sample(maybe_games, self.options.random_choice_game_count.value)
+
+        if not included_game_names:
+            raise OptionError(f"UFO 50: {self.player_name} has not selected any games.")
+
+        self.included_games = [game for name, game in ufo50_games.items() if name in included_game_names]
+
     def create_regions(self) -> None:
-        self.manager.create_regions()
+        menu = Region("Menu", self.player, self.multiworld)
+        self.multiworld.regions.append(menu)
+        for game in self.included_games:
+            game_regions = game.regions.create_regions_and_rules(self)
+            for region in game_regions.values():
+                self.multiworld.regions.append(region)
+            # !!! get menu region method
+            game_menu = self.multiworld.get_region(f"{game.game_name} - Menu", self.player)
+            menu.connect(game_menu, f"Boot {game.game_name}")
 
     def create_items(self) -> None:
-        self.manager.create_items()
+        for game in self.included_games:
+            self.multiworld.itempool += game.items.create_items(self)
 
     def get_filler_item_name(self) -> str:
-        return self.manager.get_filler_item_name()
+        return self.random.choice(self.included_games).items.get_filler_item_name()
 
     def fill_slot_data(self) -> Dict[str, Any]:
-        slot_data = self.options.as_dict()
+        slot_data = {
+            "included_games": [game.game_name for game in self.included_games]
```

Comment:
Is this player-facing at all? Because if it's not, I'd rather just get the ids in-game over the names.

Reply:
Whichever works o: I mostly didn't know what format was desired yet. I'll make it a list of IDs instead